### PR TITLE
Bluespace biobags

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Adds bluespace biobags that can store up to 100 extracts. Biobags also now can hold crossbreed extracts.

## Why It's Good For The Game

Allows to store cores and crossbreeds more conviniently.

## Changelog
:cl:
add: Added a bluespace biobag that can be printed in science protolathe
/:cl:

### CODE REVIEW

**Comment by moocowswag** [comment]
That is both such a massive buff to xenobio while also not buffing xenobio at all 90% of the time.

Its shrodingers buff

On one hand you can bomb the station even more in a single trip.

On the other hand, you only need a handful of gold / oil extracts to get shuttle called.


On one hand, you made inventory management alot easier for xenobiologists as they can fill themselves with crossbreed spells and stabilized extracts for their passives, especially in the late game.

On the other hand, even if the round lasts long enough xenobiologists tend to only make a few crossbreeds and call it a day.

**Comment by Tlaltecuhtli** [comment]
> On the other hand, you only need a dozen of gold / oil extracts to get shuttle called/kill everyone.

xenobiologist with 25 golds or oils isnt any different than one with 100

**Comment by SmArtKar** [comment]
> That is both such a massive buff to xenobio while also not buffing xenobio at all 90% of the time.
> 
> Its shrodingers buff
> 
> On one hand you can bomb the station even more in a single trip.
> 
> On the other hand, you only need a dozen of gold / oil extracts to get shuttle called/kill everyone.
> 
> On one hand, you made inventory management alot easier for xenobiologists as they can fill themselves with crossbreed spells and stabilized extracts for their passives, especially in the late game.
> 
> On the other hand, even if the round lasts long enough xenobiologists tend to only make a few crossbreeds and call it a day.

Xenobio itself should be reworked in my opinion to prevent such stuff.

**Comment by LemonInTheDark** [comment]
Why is storing more things good for the game? Outside of just buffing the role of course, why is this needed/why would I want to merge this?

**Comment by SmArtKar** [comment]
> Why is storing more things good for the game? Outside of just buffing the role of course, why is this needed/why would I want to merge this?

Well, it adds some progression to xenobio equipment as well as making game more comfortable for them. It's the same reason you want bs satchels as miner

**Comment by moocowswag** [comment]
What about making the bag a crossbreed?

**Comment by TheBonded** [comment]
Can't wait to fill a bunch of these with monkey cubes, shove em all in a locker and dump them out in evac!

**Comment by SmArtKar** [comment]
> What about making the bag a crossbreed?Probably not an option though because crossbreeds for each color are all or nothing i guess.
> 
> Also can it store beakers? If it can take a ton of bluespace beakers thats an unmatched amount of chem storage.
> 
> Well that and you can store chems in used extracts so even more chem storage.

Nothing prevents you from doing the same thing with chemical bags.

**Comment by ArcaneDefence** [comment]
I'm not a fan of reintroducing up to 100 stable rainbows immortality, like the old bluespace trashbag belt xenobiologist memes.

**Comment by TheBonded** [comment]
> I'm not a fan of reintroducing up to 100 stable rainbows immortality, like the old bluespace trashbag belt xenobiologist memes.

Stable rainbows don't stack in the same way stable metals do. If you go into crit, EVERY stable rainbow you have will all use their regen extract, which is pretty much useless

**Comment by trexter555** [comment]
i am 99% sure crossbreed cores cannot be stored in the biobag

**Comment by LemonInTheDark** [comment]
> 
> 
> > Why is storing more things good for the game? Outside of just buffing the role of course, why is this needed/why would I want to merge this?
> 
> Well, it adds some progression to xenobio equipment as well as making game more comfortable for them. It's the same reason you want bs satchels as miner

I don't mean why would a xenobio player want this, why would I want to make xenobio more powerful?
It's in a pretty strong spot as it is, outside of the subpar gameplay. It seems to me that this addition will just make it stronger then it already is, which isn't really something I want.

File: code/game/objects/items/storage/bags.dm [hunk 1]
```diff
@@ -426,6 +426,7 @@
 	worn_icon_state = "biobag"
 	desc = "A bag for the safe transportation and disposal of biowaste and other biological materials."
 	resistance_flags = FLAMMABLE
+	slot_flags = ITEM_SLOT_BELT | ITEM_SLOT_POCKETS
```
Comment (Elli-Skala) [inline code/game/objects/items/storage/bags.dm:429]: you may want to mention in the pr that you're letting bio bags also fit into pockets with this pr as well, making it up to 300 slime cores on a person at once

**Comment by YakumoChen** [comment]
> Well, it adds some progression to xenobio equipment as well as making game more comfortable for them. It's the same reason you want bs satchels as miner

Extracts alone are already equipment progression for xenobio, they don't really have a pressing need to carry more stuff other than saving a measly few clicks and making the antag even more strong.
Crossbreed holding is a absolutely massive upgrade considering storage space is the one limiting factor to passive crossbreeds' power.

File: code/game/objects/items/storage/bags.dm (hunk 1 above)
Comment (Mothblocks) [reply code/game/objects/items/storage/bags.dm:429]: or don't do this at all

File: code/game/objects/items/storage/bags.dm [hunk 2]
```diff
@@ -446,9 +447,22 @@
...
+	name = "bio bag of holding"
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bspace_biobag"
+	desc = "A bag for the safe transportation and disposal of biowaste and other biological materials."
+
+/obj/item/storage/bag/bio/holding/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_combined_w_class = INFINITY
+	STR.max_items = 100
```
Comment (Mothblocks) [inline code/game/objects/items/storage/bags.dm:464]:
```suggestion
	var/datum/component/storage/storage = GetComponent(/datum/component/storage)
	storage.max_combined_w_class = INFINITY
	storage.max_items = 100
```

**Comment by Mothblocks** [comment]
> Why is storing more things good for the game? Outside of just buffing the role of course, why is this needed/why would I want to merge this?

+1 to this

**Comment by Mothblocks** [comment]
I don't see a compelling reason to add this in the state its in.

**Pull request closed without merging**